src/radical/pilot/compute_pilot.py

```diff
@@ -564,35 +579,48 @@ def cancel(self):
 
     # --------------------------------------------------------------------------
     #
-    def stage_in(self, directives):
+    def stage_in(self, sds):
         '''
         Stages the content of the staging directive into the pilot's
         staging area
         '''
 
-        # This staging request is actually served by the pmgr *launching*
-        # component, because that already has a channel open to the target
-        # resource which we can reuse.  We might eventually implement or
-        # interface to a dedicated data movement service though.
+        sds = ru.as_list(sds)
+
+        for sd in sds:
+            sd['prof_id'] = self.uid
 
-        # send the staging request to the pmg launcher
-        self._pmgr._pilot_staging_input(self.as_dict(), directives)
+        for sd in sds:
+            sd['source'] = str(complete_url(sd['source'], self._loc_ctx, self._log))
+            sd['target'] = str(complete_url(sd['target'], self._rem_ctx, self._log))
+
+        # ask the pmgr to send the staging reuests to the stager
+        self._pmgr._pilot_staging_input(sds)
 
 
     # --------------------------------------------------------------------------
     #
-    def stage_out(self):
+    def stage_out(self, sds=None):
         '''
-        fetch `staging_output.tgz` from the pilot sandbox, and store in $PWD
+        fetch files (default:`staging_output.tgz`) from the pilot sandbox
         '''
 
-        try:
-            psbox = self._session.get_fs_dir(self._pilot_sandbox)
-            psbox.copy('staging_output.tgz', self._client_sandbox)
+        sds = ru.as_list(sds)
+
+        if not sds:
+             sds = [{'source': 'client:///staging_output.tgz',
```

Comment:
should here be `source`: `pilot...` and `target`: `client...`?

Reply:
Indeed - thanks!